mbof/views.py

```diff
@@ -28,25 +31,27 @@ class EventViewSet(viewsets.ModelViewSet):
     """
     queryset = Event.objects.all().order_by('postingTime')
     serializer_class = EventSerializer
-    
-
-class CurrentEventViewSet(EventViewSet):
-    """
-    API endpoint that allows users to view current events in db.
-    """
-    def get_queryset(self):
-        current_time = timezone.now()
-        return Event.objects.filter(startTime__lte=current_time, endTime__gte=current_time).order_by('postingTime')
 
 
-class UpcomingEventViewSet(EventViewSet):
+class EventListViewSet(generics.ListAPIView):
     """
-    API endpoint that allows users to view upcoming events in db.
+    API endpoint that allows users to be viewed or edited.
     """
+    serializer_class = EventSerializer
+
     def get_queryset(self):
-        current_time = timezone.now()
-        one_week_out = current_time + datetime.timedelta(days=7)
-        return Event.objects.filter(startTime__range=[current_time, one_week_out]).order_by('postingTime')
+        logger.debug('Type=' + self.kwargs['type'])
+        query_type = self.kwargs['type']
```

Comment:
Check whether `type` exists or has value before assigning its value to `query_type`.